### PR TITLE
Pre-oop Done

### PR DESCRIPTION
**Pull request by CrazyPike**
### Pre-oop

[Demo](
https://crazypike.github.io/frontend-2019-homeworks/submissions/CrazyPike/pre-oop%237/index.html) |
[Code base](https://github.com/CrazyPike/frontend-2019-homeworks/tree/master/submissions/CrazyPike/pre-oop%237)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/CrazyPike/pre-oop#7/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
```
Comment (OleksiyRudenko) [inline submissions/CrazyPike/pre-oop#7/index.js]: This row exposes way too basic skills. Please, remove.

File: submissions/CrazyPike/pre-oop#7/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+print(`${dog.species}; ${dog.name}; ${dog.gender}; ${dog.legs}; ${dog.hands}; ${dog.phrase}`);
```
Comment (OleksiyRudenko) [inline submissions/CrazyPike/pre-oop#7/index.js]: There is a pattern here. Please, implement a function that would take an object as an argument and return a formatted string where components are combined employing `Array.join`.

File: submissions/CrazyPike/pre-oop#7/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+print(`${cat.species}; ${cat.name}; ${cat.gender}; ${cat.legs}; ${cat.hands}; ${cat.phrase}`);
+print(`${man.species}; ${man.name}; ${man.gender}; ${man.legs}; ${man.hands}; ${man.phrase}`);
+print(`${woman.species}; ${woman.name}; ${woman.gender}; ${woman.legs}; ${woman.hands}; ${woman.phrase}`);
```
Comment (OleksiyRudenko) [inline submissions/CrazyPike/pre-oop#7/index.js]: There is a pattern in the above series of `print`. Please, refactor.
Hint:
```
[varA, varB].forEach(item => someFunction(item));
```

**Comment by OleksiyRudenko** [comment]
@CrazyPike Please, also remove all files except `index.js` as it is sufficient to conduct the code review.

File: submissions/CrazyPike/pre-oop#7/index.js (hunk 3 above)
Comment (CrazyPike) [reply submissions/CrazyPike/pre-oop#7/index.js]: if I got you right please see the next commit

**Comment by CrazyPike** [comment]
I left only index.js in the next commit. Is it enough?

**Comment by OleksiyRudenko** [comment]
@CrazyPike We still have 39 files in the PR.
You need to employ `git rm -rf` to untrack and remove files.

File: submissions/CrazyPike/pre-oop#7/index.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    species: 'human',
+    legs: 2,
+    hands: 2,
+    name: 'Jack',
+    gender: 'female',
+    phrase: 'Jenny'
+}
+
+function stringify (obj) {
+    return Object.values(obj).join(';');
```
Comment (OleksiyRudenko) [inline submissions/CrazyPike/pre-oop#7/index.js]: In your past implementation you had a specific order of props that also matched tech spec.

Please, note that order of props in an object is not guaranteed. So neither `for ... in`, nor `Object` methods guarantee as well.
Please, list properties explicitly and apply `.join()`.

Comment (CrazyPike) [reply submissions/CrazyPike/pre-oop#7/index.js]: Thx, please see the solution.

**Comment by CrazyPike** [comment]
thanks for help!

**Pull request merged**